components/script/textinput.rs

```diff
@@ -370,27 +469,30 @@ impl<T: ClipboardProvider> TextInput<T> {
 
             max_length - len_after_selection_replaced
         } else {
-            usize::MAX
+            UTF16CodeUnits(usize::MAX)
         };
 
-        let last_char_index = len_of_first_n_code_units(&*insert, allowed_to_insert_count);
-        let chars_to_insert = &insert[..last_char_index];
+        let ByteOffset(last_char_index) =
+            len_of_first_n_code_units(&*insert, allowed_to_insert_count);
+        let bytes_to_insert = &insert[..last_char_index];
```

Comment:
Nit: neither `chars_to_insert` nor `bytes_to_insert` to quite the right name for a variable of type `&str`. (Bytes would be `&[u8]`.) Maybe just `to_insert`?